lib/declarative_authorization/controller/grape.rb

```diff
@@ -0,0 +1,113 @@
+require File.dirname(__FILE__) + '/../authorization.rb'
+require File.dirname(__FILE__) + '/dsl.rb'
+require File.dirname(__FILE__) + '/runtime.rb'
+
+#
+# This mixin can be used to add declarative authorization support to APIs built using Grape
+# https://github.com/ruby-grape/grape
+#
+# Usage:
+#   class MyApi < Grape::API
+#     include Authorization::Controller::Grape
+#
+#     get :hello do
+#     end
+#   end
+#
+# NOTE: actions in authorization rules must be named `{METHOD} {URL}`. eg
+#   has_permission_on :my_api, to: 'GET /my_api/hello'
+#
+module Authorization
+  module Controller
+    module Grape
+      def self.included(base) # :nodoc:
+        base.extend ClassMethods
+
+        base.extend ::Authorization::Controller::DSL
+
+        base.module_eval do
+          add_filter!
+        end
+
+        base.helpers do
+          include ::Authorization::Controller::Runtime
+
+          def authorization_engine
+            ::Authorization::Engine.instance
+          end
+
+          def filter_access_filter # :nodoc:
+            action_name = "#{request.request_method} #{route.origin}"
+
+            permissions = api_class.all_filter_access_permissions
+            all_permissions = permissions.select {|p| p.actions.include?(:all)}
+            matching_permissions = permissions.select {|p| p.matches?(action_name)}
+            allowed = false
+            auth_exception = nil
+
+            begin
+              allowed = if !matching_permissions.empty?
```

Comment:
Could we use `.present?` instead

Reply:
If we do then it should be changed in both the Rails and Grape versions in a new commit (code was duplicated). 😄

Reply:
This was existing code taken from `lib/declarative_authorization/in_controller.rb` that also exists in `lib/declarative_authorization/controller/rails.rb`.

For the sake of keeping this PR smaller I chose not to clean up existing code.

Reply:
@dvicory good point about code duplication. Perhaps further refactoring is needed.